openedx_caliper_tracking/tests/tests.py

```diff
@@ -81,11 +81,43 @@ class CaliperDeliveryTestCase(TestCase):
     )
     def test_event_is_not_called_without_settings(self, delivery_mock):
         """
-        If the settings CALIPER_DELIVER_ENDPOINT and CALIPER_DELIVERY_AUTH_TOKEN are
-        not present then the API call shouldn't fire
+        If the settings ENABLE_CALIPER_EVENTS_DELIVERY, CALIPER_DELIVER_ENDPOINT
+        and CALIPER_DELIVERY_AUTH_TOKEN are not present then the API call shouldn't fire
         """
-        del settings.CALIPER_DELIVERY_ENDPOINT
-        del settings.CALIPER_DELIVERY_AUTH_TOKEN
+        if settings.FEATURES.get('ENABLE_CALIPER_EVENTS_DELIVERY'):
```

Comment:
Please use patch.dict for changing the dict values.
https://docs.python.org/3/library/unittest.mock.html#patch-dict

Reply:
done.

Reply:
In this scenario, it is not possible but I have used it wherever it is possible.